lib/src/main/scala/code/chymyst/jc/Reaction.scala

```diff
@@ -190,10 +202,10 @@ final case class OutputMoleculeInfo(molecule: Molecule, flag: OutputPatternType)
 }
 
 // This class is immutable.
-final case class ReactionInfo(inputs: List[InputMoleculeInfo], outputs: List[OutputMoleculeInfo], guardPresence: GuardPresenceFlag, sha1: String) {
+final case class ReactionInfo(inputs: Array[InputMoleculeInfo], outputs: Array[OutputMoleculeInfo], guardPresence: GuardPresenceFlag, sha1: String) {
 
   // The input pattern sequence is pre-sorted for further use.
-  private[jc] val inputsSorted: List[InputMoleculeInfo] = inputs.sortBy { case InputMoleculeInfo(mol, flag, sha) =>
+  private[jc] val inputsSorted: List[InputMoleculeInfo] = inputs.sortBy { case InputMoleculeInfo(mol, _, flag, sha) =>
```

Comment:
No need for IndexedSeq type for inputsSorted then? I assume you'll only need to traverse sequentially. If you start with Array inputs you may as well stick with Array all the way and not degenerate to inferior List container.